src/System.Globalization/tests/DateTimeFormatInfo/CalendarData.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Reflection;
+using Xunit;
+
+namespace System.Globalization.Tests
+{
+    public class CalendarData
+    {
+        private static readonly Type calendarDataType = Type.GetType("System.Globalization.CalendarData");
+        private static readonly MethodInfo normalizeDatePattern = calendarDataType.GetTypeInfo().GetDeclaredMethod("NormalizeDatePattern");
+
+        [Theory]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        [InlineData("'EEEE '' LLL' y", "'EEEE '' LLL' yyyy")]
+        [InlineData("EEEE, d MMMM y", "dddd, d MMMM yyyy")]
+        [InlineData("ee, d LLLLL", "ddd, d MMM")]
+        [InlineData("ccccc, d", "ddd, d")]
+        [InlineData("GG yy/M/d", "g yy/M/d")]
+        public static void TestNormalizeDatePattern(string input, string expected)
```

Comment:
I did it that way to start, but then decided against it.  The reason is because a lot of those cultures' formats have changes between Win8.1, Win10, and the version of CLDR that Ubuntu 14.04 uses.  So I figured hard-coding that much data that could/will change often seems like a maintenance hassle.  In order to cover all those cases, I would need to use 4-5 different cultures.  This way, the culture data can change, but the tests stay resilient.  The tests for en-US and es-ES test to make sure we are pulling the data correctly.